test/variant_sender_test.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <unifex/defer.hpp>
+#include <unifex/dematerialize.hpp>
+#include <unifex/just.hpp>
+#include <unifex/just_error.hpp>
+#include <unifex/just_done.hpp>
+#include <unifex/materialize.hpp>
+#include <unifex/variant_sender.hpp>
+#include <unifex/scheduler_concepts.hpp>
+#include <unifex/sync_wait.hpp>
+#include <unifex/timed_single_thread_context.hpp>
+
+#include <chrono>
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+using namespace unifex;
+using namespace std::chrono;
+using namespace std::chrono_literals;
+
+struct IntAndStringReceiver {
+    void set_value(int) noexcept {
+
+    }
+
+    void set_value(std::string) noexcept {
+
+    }
+
+    void set_done() noexcept {}
+
+    void set_error(std::exception_ptr) noexcept {}
+};
+
+template<bool lvalueConnectNoexcept = true, bool rvalueConnectNoexcept = true>
+struct TestSender {
+  template <
+    template <typename...> class Variant,
+    template <typename...> class Tuple>
+  using value_types = Variant<std::string>;
```

Comment:
I think you need to use `Tuple` somewhere to make this compatible with all the type machinery, no?
```suggestion
  using value_types = Variant<Tuple<std::string>>;
```